rust/agama-network/src/nm/dbus.rs

```diff
@@ -0,0 +1,443 @@
+//! This module implements some functions to convert from/to D-Bus types
+//!
+//! Working with hash maps coming from D-Bus is rather tedious and it is even worse when working
+//! with nested hash maps (see [NestedHash] and [OwnedNestedHash]).
+use super::model::*;
+use crate::model::*;
+use std::collections::HashMap;
+use std::net::Ipv4Addr;
+use uuid::Uuid;
+use zbus::zvariant::{self, Value};
+
+const ETHERNET_KEY: &str = "802-3-ethernet";
+const WIRELESS_KEY: &str = "802-11-wireless";
+const WIRELESS_SECURITY_KEY: &str = "802-11-wireless-security";
+const LOOPBACK_KEY: &str = "loopback";
+
+/// Nested hash to send to D-Bus.
+type NestedHash<'a> = HashMap<&'a str, HashMap<&'a str, zvariant::Value<'a>>>;
+/// Nested hash as it comes from D-Bus.
+type OwnedNestedHash = HashMap<String, HashMap<String, zvariant::OwnedValue>>;
```

Comment:
btw I think this types can be useful even in agama lib as we use hashes also CLI and I fight with it a bit.